### PR TITLE
Update docker files.

### PR DESCRIPTION
**Pull request by frankfliu**
1. Remove unused python package.
2. Use python3 in docker container.
3. Remove unnecessary packages, this will reduce 800M image size.
4. Install nightly build of mxnet.
5. Upgrade mxnet to 1.3.1

Before or while filing an issue please feel free to join our [ slack channel](https://join.slack.com/t/mms-awslabs/shared_invite/enQtNDk4MTgzNDc5NzE4LTBkYTAwMjBjMTVmZTdkODRmYTZkNjdjZGYxZDI0ODhiZDdlM2Y0ZGJiZTczMGY3Njc4MmM3OTQ0OWI2ZDMyNGQ) to get in touch with development team, ask questions, find out what's cooking and more!

## Testing done:

**To run CI tests on your changes refer [README.md](https://github.com/awslabs/mxnet-model-server/blob/master/ci/README.md)**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docker/Dockerfile.cpu [hunk 1]
```diff
@@ -3,23 +3,25 @@ FROM ubuntu:18.04
...
-    build-essential \
-    libatlas-base-dev \
-    libopencv-dev \
-    graphviz \
-    python-dev \
+    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    fakeroot \
+    ca-certificates \
+    dpkg-dev \
+    g++ \
```
Comment (vdantu) [inline docker/Dockerfile.cpu]: Why do we need this?

File: docker/Dockerfile.cpu [hunk 2]
```diff
@@ -3,23 +3,25 @@ FROM ubuntu:18.04
...
     openjdk-8-jdk-headless \
     curl \
     vim \
     && rm -rf /var/lib/apt/lists/* \
     && cd /tmp \
     && curl -O https://bootstrap.pypa.io/get-pip.py \
-    && python get-pip.py
+    && python3 get-pip.py
+
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 1
```
Comment (vdantu) [inline docker/Dockerfile.cpu]: Why do we need update-alternatives?

File: docker/Dockerfile.nightly-cpu [hunk 3]
```diff
@@ -3,20 +3,22 @@ FROM ubuntu:18.04
...
     && rm -rf /var/lib/apt/lists/* \
     && cd /tmp \
     && curl -O https://bootstrap.pypa.io/get-pip.py \
-    && python get-pip.py
+    && python3 get-pip.py
+
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 1
+RUN update-alternatives --install /usr/local/bin/pip pip /usr/local/bin/pip3 1
 
 RUN pip install --no-cache-dir --pre mxnet-model-server \
```
Comment (vdantu) [inline docker/Dockerfile.nightly-cpu]: dont we need pip3 to install these packages?

File: docker/Dockerfile.cpu (hunk 1 above)
Comment (frankfliu) [reply docker/Dockerfile.cpu]: This is much smaller than build-enssential

File: docker/Dockerfile.cpu (hunk 2 above)
Comment (frankfliu) [reply docker/Dockerfile.cpu]: python3-dev only install python3 command line, there is no /usr/bin/python exists

File: docker/Dockerfile.nightly-cpu (hunk 3 above)
Comment (frankfliu) [reply docker/Dockerfile.nightly-cpu]: I created alternative for pip -> pip3

File: docker/Dockerfile.cpu (hunk 2 above)
Comment (vdantu) [reply docker/Dockerfile.cpu]: I think I should understand this tool well. So, is this just creating a sym_link?

File: docker/Dockerfile.cpu (hunk 1 above)
Comment (vdantu) [reply docker/Dockerfile.cpu]: Did we need build-essential for opencv? Isn't g++ a compiler? Would we need it?

**Pull request merged**